### PR TITLE
Issue #221 added api/v1/users documentation 

### LINKED ISSUE #221: Add API Documentation for Users endpoints
# Feature
## Why is this feature being added?

Our contributors need to be able to research, confirm and build against the API endpoints that the backend offers.

## What should your feature do?
Document the following backend API endpoints in our [online API documentation](https://operationcodeapi.docs.apiary.io/#).  When doing so, **please follow** the steps described in our `README` for [how to document our API](https://github.com/OperationCode/operationcode_backend#api-documentation).

The endpoints to be documented are:

- [ ] `api/v1/users_controller#create`
- [ ] `api/v1/users_controller#update`
- [ ] `api/v1/users_controller#verify`

### PR DESCRIPTION
**Pull request by wimo7083**
# Description of changes
Included 3 new Apiary API routes. 
POST  `api/v1/users_controller#create` 
PATCH `api/v1/users_controller#update`
POST for `api/v1/users_controller#verify`

# Issue Resolved
Resolved issue of lacking API docs for users.
Fixes #221

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
 
 API endpoints that Operation Code's Rails backend makes available to its React frontend.
 
+## Users | Create [/api/v1/users{?first_name,last_name,email,zip,password,mentor}]
```
Comment (hpjaj) [inline apiary.apib]: Can you please alphabetize the placement of these in the file.  First by model (i.e. `User`), then by action (i.e. `create`)

File: apiary.apib [hunk 2]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+  + mentor (boolean, required) - Is User requesting mentor in the form of a boolean
+
+### Create new user record [POST]
+
++ Request (application/json)
+
+    + Body
+
+            {            
+            first_name: "Mike",
```
Comment (hpjaj) [inline apiary.apib]: Can you pls add indenting in keeping with other `Body` hashes

File: apiary.apib [hunk 3]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+            zip: "80020" ,        
+            password: "hunter2",    
+            mentor: "true",   
+            }
+
+
++ Response 201 (application/json)
+
+        {
+        token: Bearer Access-Token
```
Comment (hpjaj) [inline apiary.apib]: Indenting

File: apiary.apib [hunk 4]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+        }
+
++ Response 422 (application/json)
+
+        {
+        errors: "Some error message"
```
Comment (hpjaj) [inline apiary.apib]: Indenting

File: apiary.apib [hunk 5]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+
+    + Headers
+
+            Authorization: Bearer Access-Token,
+            verified: true
+
+    + Body
+
+            {
+            verified: true
```
Comment (hpjaj) [inline apiary.apib]: See above re: `access_token`

File: apiary.apib [hunk 6]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+
+        {
+        errors: "Some error message"
+        }
+
+## Users | Update [/api/v1/users{?education_level,mentor,scholarship_info,employment_status,company_name,company_role,volunteer,branch_of_service,interests}]
+
++ Parameters
+
+  + education_level (string, required) - Users highest level of education in the form of a string
```
Comment (hpjaj) [inline apiary.apib]: These are all optional (vs. required)

File: apiary.apib [hunk 7]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
++ Response 200 (application/json)
+
+    + Body
+
+            {
+            first_name: "Mike",
+            last_name: "Johnson",
+            email: "dreams@operationcode.org",     
+            zip: "80020",        
+            password: "hunter2",
```
Comment (hpjaj) [inline apiary.apib]: Technically `encrypted_password` is what is returned, but you can leave it out all together.

File: apiary.apib [hunk 8]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+            errors: "Some error message"
+            }
+    
+## Users | Verify [/api/v1/users/profile/verify{?verified}]
+
++ Parameters
+
+  + verified (boolean, required) - Users verified veteran status in the form of a boolean
+
+### Modify User record verified status [POST]
```
Comment (hpjaj) [inline apiary.apib]: Perhaps something like, "Verifies a user's military affiliation status through ID.me"

File: apiary.apib [hunk 9]
```diff
@@ -5,6 +5,148 @@ HOST: https://operationcode.org/
...
+## Users | Create [/api/v1/users{?first_name,last_name,email,zip,password,mentor}]
+
++ Parameters
+
+  + first_name (string, required) - First Name of the user in the form of an string
+  + last_name (string, required) - Last Name of the user in the form of an string
+  + email (string, required) - Email of the User in the form of a string
+  + zip (string, required) - Zipcode of the User in the form of a string
+  + password (string, required) - Password of the User in the form on a string
+  + mentor (boolean, required) - Is User requesting mentor in the form of a boolean
```
Comment (hpjaj) [inline apiary.apib]: I believe `mentor` is optional

Comment (wimo7083) [reply apiary.apib]: I think you're [right]( https://github.com/OperationCode/operationcode_frontend/blob/bff605dc75ffeca42434d168145c40c89cceeb96/src/scenes/home/signup/signup.js#L72) from a front end perspective.

File: apiary.apib (hunk 1 above)
Comment (wimo7083) [reply apiary.apib]: Done, also adjusted `Location`

File: apiary.apib (hunk 2 above)
Comment (wimo7083) [reply apiary.apib]: Fixed.

File: apiary.apib (hunk 3 above)
Comment (wimo7083) [reply apiary.apib]: Fixed.

File: apiary.apib (hunk 6 above)
Comment (wimo7083) [reply apiary.apib]: So if they submit no [params](https://github.com/OperationCode/operationcode_backend/blob/50aaf886e8ddda7a6ea05bbac4ba496df94ebc96/app/controllers/api/v1/users_controller.rb#L24) it returns a 422 or 200?

File: apiary.apib (hunk 7 above)
Comment (wimo7083) [reply apiary.apib]: Fixed.

File: apiary.apib (hunk 8 above)
Comment (wimo7083) [reply apiary.apib]: Fixed.

File: apiary.apib (hunk 5 above)
Comment (wimo7083) [reply apiary.apib]: Removed verified from both header and body, and fixed indentation.

File: apiary.apib (hunk 4 above)
Comment (wimo7083) [reply apiary.apib]: Fixed.

**Comment by wimo7083** [comment]
Completed changes you requested. One thing I noticed was how some of the endpoints had combined grouping. Codeschool | Show or Delete is one example. 

There's inconsistent use of Body in the previous documentation. But apiary doesn't seem to mind. Not sure of your thoughts. 

```
## User | Reset Password [/api/v1/users/passwords/reset{?email}]

+ Parameters

    + email (string, required) - The email address of the requesting User

### Reset the User's Password  [POST]

+ Request (application/json)

    + Body

            {
                "email": "frank@example.com"
            }

+ Response 200 (application/json)

    + Body

            {
                status: "ok"
            }

+ Response 422 (application/json)

        {
            errors: "Some error message"
        }
```
```
### Delete an Existing CodeSchool [DELETE]

  + Request (application/json)

      + Headers

              Authorization: Bearer Access-Token

      + Body

              {
                  "id": 1
              }

  + Response 200 (application/json)

      + Body

              {
                  status: "ok"
              }

  + Response 422 (application/json)

      + Body

              {
                  errors: "Some error message"
              }
```

File: apiary.apib [hunk 10]
```diff
@@ -987,6 +990,43 @@ API endpoints that Operation Code's Rails backend makes available to its React f
...
+  + mentor (boolean, optional) - Is User requesting mentor in the form of a boolean
+
+### Create new user record [POST]
+
++ Request (application/json)
+
+    + Body
+
+            {
+            first_name: "Mike",
```
Comment (hpjaj) [inline apiary.apib]: Indent :)

File: apiary.apib [hunk 11]
```diff
@@ -1040,6 +1080,108 @@ API endpoints that Operation Code's Rails backend makes available to its React f
...
+
+            {
+                errors: "Some error message"
+            }
+
+## User | Verify [/api/v1/users/profile/verify{?verified}]
+
++ Parameters
+
+  + verified (boolean, required) - User has verified military affiliation status through ID.me
```
Comment (hpjaj) [inline apiary.apib]: The `param` here is only [access_token](https://github.com/OperationCode/operationcode_backend/blob/master/app/controllers/api/v1/users_controller.rb#L33)

File: apiary.apib [hunk 12]
```diff
@@ -1040,6 +1080,108 @@ API endpoints that Operation Code's Rails backend makes available to its React f
...
++ Request (application/json)
+
+    + Headers
+
+            Authorization: Bearer Access-Token
+
+    + Body
+
+            {
+                verified: true
```
Comment (hpjaj) [inline apiary.apib]: Body is only the `access_token`

File: apiary.apib [hunk 13]
```diff
@@ -461,6 +461,39 @@ API endpoints that Operation Code's Rails backend makes available to its React f
           {
               errors: "Some error message"
           }
+
+## Location | Delete [/api/v1/code_schools/:code_school_id/locations/:id]
+
+### Delete an Existing Location [DELETE]
```
Comment (hpjaj) [inline apiary.apib]: Nice 👍

File: apiary.apib [hunk 14]
```diff
@@ -1040,6 +1080,108 @@ API endpoints that Operation Code's Rails backend makes available to its React f
...
+
+  + verified (boolean, required) - User has verified military affiliation status through ID.me
+
+### Modify User record verified status [POST]
+
++ Request (application/json)
+
+    + Headers
+
+            Authorization: Bearer Access-Token
```
Comment (hpjaj) [inline apiary.apib]: This method actually does not require authorization in the headers.  Only the [update](https://github.com/OperationCode/operationcode_backend/blob/master/app/controllers/api/v1/users_controller.rb#L4) method does.

Comment (wimo7083) [reply apiary.apib]: Fixed.

**Comment by wimo7083** [comment]
Made some changes, also found some instances where  Param items only had 2 spaces.

**Pull request merged**